storage/innobase/fsp/fsp0file.cc

```diff
@@ -687,7 +689,7 @@ Datafile::validate_first_page(lsn_t*	flush_lsn,
 				ut_free(m_encryption_iv);
 				m_encryption_key = NULL;
 				m_encryption_iv = NULL;
-				return(DB_CORRUPTION);
+				return(DB_IO_DECRYPT_FAIL);
```

Comment:
I feel it is better to add to invalid encrypted table list here.

Reply:
I wasn´t sure if it was ok to add a new return type. Shall I use same as 8.0 https://github.com/percona/percona-xtrabackup/blob/8.0/storage/innobase/fsp/fsp0file.cc#L703 here ?

Reply:
no need to change return type. Just add to vector here.

Reply:
Yes, makes (wayyy more)  sense. Will fix

Reply:
but the caller would abort I guess. So maybe you are right :laughing:

Reply:
We can return DB_PAGE_IS_BLANK which will make it continue to operate if we are in the backup phase, and validate_missing_encryption_tablespaces will ensure that we have parsed proper encryption information by the end of backup.

Reply:
and we really have an all-zero page right (except page_id)? in that case, DB_PAGE_IS_BLANK makes sense

Reply:
Correct, the flags that are written syncronously are set (page_id, encryption flag, ...), and the ones that are written async (encryption data, ...) are empty. DB_PAGE_IS_BLANK will indeed make it clear.

Reply:
Cool! :+1: